bytecode/instr.py

```diff
@@ -282,6 +282,37 @@ def stack_effect(self, jump=None):
         else:
             return dis.stack_effect(self._opcode, arg, jump=jump)
 
+    def pushes_back(self):
+        opname = _opcode.opname[self._opcode]
+        if opname in ["CALL_FINALLY"]:
+            return False
+        return (
+            opname.startswith("UNARY_")
+            or opname.startswith("GET_")
+            or opname.startswith("BINARY_")
+            or opname.startswith("INPLACE_")
+            or opname.startswith("BUILD_")
+            or opname.startswith("CALL_")
+        ) or opname in [
+            "LIST_TO_TUPLE",
+            "LIST_EXTEND",
+            "SET_UPDATE",
+            "DICT_UPDATE",
+            "DICT_MERGE",
+            "IS_OP",
+            "CONTAINS_OP",
+        ]
+
+    def pre_and_post_stack_effect(self, jump=None):
+
+        _effect = self.stack_effect(jump=jump)
+        # To compute pre size and post size to avoid segfault cause by not enough stack element
+        if _opcode.opname[self._opcode].startswith("DUP_TOP"):
```

Comment:
We may want to handle the ROT_ opcodes in the same way since I assume the interpreter crashes if there are not enough objects on the stack.